src/components/SharedComponents/VideoPlayer/useMediaClipThumbnails.ts

```diff
@@ -0,0 +1,27 @@
+import { PlaybackPolicy, useSignedThumbnailToken } from "./useSignedVideoToken";
+
+interface UseThumbnailImageProps {
+  playbackId: string;
+  playbackPolicy: PlaybackPolicy;
+  isLegacy?: boolean;
+}
+
+const useMediaClipThumbnail = ({
+  playbackId,
+  playbackPolicy,
+  isLegacy = false,
+}: UseThumbnailImageProps) => {
+  const thumbnailToken = useSignedThumbnailToken({
+    playbackId,
+    playbackPolicy,
+    isLegacy,
+  });
+
+  const thumbnailImage = thumbnailToken
```

Comment:
check if there is `thumbnailToken.playbackToken` first, the `thumbailToken` returns {loading: true, tumbnailToken: null` first so it will cause the url to be wrong and that's what causing flicker when you first load the page (when the image looks broken)

I am pretty sure I fixed it in my branch so maybe it is worth merging main in first?